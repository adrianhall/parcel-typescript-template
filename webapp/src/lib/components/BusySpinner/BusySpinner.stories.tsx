import React from 'react';
import { addDecorator } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import BusySpinner from '.';

export default {
  title: 'lib/components/BusySpinner',
  component: BusySpinner,
  decorators: [ withKnobs ]
};

const rootStyle: React.CSSProperties = {
  backgroundColor: '#F0F0F0',
  color: '#000000',
  padding: '1rem'
};
addDecorator((storyFn) => (<div style={rootStyle}>{storyFn()}</div>));

/* Story #1 - the component is busy */
export const normal: React.SFC<{}> = () => (
  <BusySpinner isBusy={boolean('isBusy', false)} onClick={action('onClick')}>
    This is some content
  </BusySpinner>
);
