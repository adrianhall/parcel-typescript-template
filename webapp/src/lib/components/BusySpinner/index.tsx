import React from 'react';

/**
 * Properties for the BusySpinner component
 *
 * @interface Props
 */
interface Props {
  /** Children to be displayed if the component is not busy */
  children: React.ReactNode;
  /** Determines if the component is considered busy */
  isBusy?: boolean;
  /** Event handler for the onClick event */
  onClick?: (event: React.MouseEvent<HTMLSpanElement>) => void;
}


/**
 * Displays a busy spinner instead of the content if the component is busy.
 *
 * @param props.children to be displayed if the component is not busy
 * @param props.isBusy = false, determines if the component is busy
 */
const BusySpinner: React.SFC<Props>
  = ({ children, isBusy = false, onClick }: Props) => {
    if (isBusy) {
      return (<span data-id="spinner" onClick={onClick}>I'm busy</span>);
    }

    return (<>{children}</>);
  };

export default BusySpinner;
