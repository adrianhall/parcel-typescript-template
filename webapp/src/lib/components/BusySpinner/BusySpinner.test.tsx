/* eslint-env jest */
import { shallow, render } from 'enzyme';
import React from 'react';
import BusySpinner from '.';

describe('BusySpinner', () => {
  it('spins busy when isBusy is true', () => {
    const actual = shallow(
      <BusySpinner isBusy>
        <p data-id="test-data">Test Data</p>
      </BusySpinner>
    );
    expect(actual).toMatchSnapshot();
  });

  it('displays the content when isBusy is false', () => {
    const actual = render(
      <BusySpinner>
        <p data-id="test-data">Test Data</p>
      </BusySpinner>
    );
    const content = actual.find('[data-id="test-data"]');
    expect(content).toBeDefined();
  });
});
