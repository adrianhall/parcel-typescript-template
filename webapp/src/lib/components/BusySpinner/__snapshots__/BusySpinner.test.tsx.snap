// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusySpinner spins busy when isBusy is true 1`] = `
<span
  data-id="spinner"
>
  I'm busy
</span>
`;
